### PR TITLE
Moderator UI for editing channel description

### LINKED ISSUE #263: Moderation: About this channel
As a channel moderator, I'd like to display some information about the channel, such as rules for posting, along side the list of posts. 

### acceptance criteria
- [ ] Store description in reddit, if possible <- let's discuss before implementing
- [ ] Use existing PRAW APIs, if possible

Be sure to discuss the approach on this ticket before implementing.

### LINKED ISSUE #357: Channel API should allow moderators to update the channel
Currently the API only allows users marked as "staff" (i.e. the API user) to modify the channel. We need to allow moderators to do this to support #263

### PR DESCRIPTION
**Pull request by rhysyngsun**
#### What are the relevant tickets?
Fixes #263 
Fixes #357 

#### What's this PR do?
Adds the UI to update the channel description for channel moderators

#### How should this be manually tested?
- Go to a channel page as a regular user, you should see no difference in UI from master
- Go to a channel page as a moderator, you should see an edit button above the channel description.
- Click the edit button and verify you can edit the channel description and the channel page updates to reflect that.

### CODE REVIEW

File: static/js/components/ChannelSidebar.js [hunk 1]
```diff
@@ -1,21 +1,32 @@
...
+const ChannelSidebar = ({ channel, isModerator }: ChannelSidebarProps) => {
   if (!channel) {
     return null
   }
   return (
     <Card title="About this channel" className="channel-about">
+      {isModerator
+        ? <div className="edit-button">
+          <Link to={editChannelURL(channel.name)}>
+            <i className="material-icons edi">edit</i>
```
Comment (aliceriot) [inline static/js/components/ChannelSidebar.js]: is `.edi` a typo here?

File: static/js/components/admin/ChannelCreateForm.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
+// @flow
+import React from "react"
+
+import { CHANNEL_TYPE_PUBLIC, CHANNEL_TYPE_PRIVATE } from "../../lib/channels"
+
+import type { ChannelForm } from "../../flow/discussionTypes"
+
+export default class ChannelEditForm extends React.Component<*, void> {
```
Comment (aliceriot) [inline static/js/components/admin/ChannelCreateForm.js]: should this class be called `ChannelCreateForm`?

File: static/js/components/admin/ChannelEditForm.js [hunk 3]
```diff
@@ -1,82 +1,61 @@
...
+import { validationMessage } from "../../lib/validation"
 
-import type { ChannelForm } from "../../flow/discussionTypes"
+import type {
+  ChannelForm,
+  ChannelEditValidation
+} from "../../flow/discussionTypes"
 
-export default class ChannelEditForm extends React.Component<*, void> {
+export default class ChannelCreateForm extends React.Component<*, void> {
```
Comment (aliceriot) [inline static/js/components/admin/ChannelEditForm.js]: Should this one be called `ChannelEditForm`?

File: static/js/lib/validation.js [hunk 4]
```diff
@@ -65,3 +66,18 @@ export const validatePostCreateForm = validate([
...
   postUrlOrTextPresent
 ])
+
+export const validateChannelEditForm = validate([
+  validation(
+    R.compose(R.lte(5120), R.length),
+    R.lensPath(["value", "description"]),
+    "Description length is limited to 5120 characters"
+  )
+])
```
Comment (aliceriot) [inline static/js/lib/validation.js]: might make sense to add a very basic test for this

File: channels/api.py [hunk 5]
```diff
@@ -714,17 +714,32 @@ def remove_moderator(self, moderator_name, channel_name):
...
+
+        Args:
+            channel_name(str): name of the channel
+            moderator_name(str): moderator username
+
+        Returns:
+            bool: True if the given username is a moderator on the channel
+        """
+        # generators always eval True, so eval as list first and then as bool
+        return bool(list(self.list_moderators(channel_name, moderator_name=moderator_name)))
```
[the remark this answers is not in the document]
Comment (rhysyngsun) [reply channels/api.py]: Yup, pylint actually enforces truthiness of the list over checking `len() == 0`, so this is preferred.

**Comment by rhysyngsun** [comment]
@aliceriot feedback addressed

**Pull request merged**